### PR TITLE
Feature/show gallows img

### PR DESCRIPTION
**Pull request by DominikDR**
This pull request contains functionalities and features of another two branches:

- feature/phrase-encrypt
- feature/show-game-over

Please checkout it and then I would like merge them to the master.

### CODE REVIEW

**Comment by Dammic** [comment]
You should rename the files names from '1' to something more descriptive.

File: src/components/Alphabet/Alphabet.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @media (min-width: 576px) {
+        grid-template-columns: repeat(auto-fill, minmax(45px, 1fr));
+        grid-auto-rows: 45px;
+        grid-gap: 12px;
+    }
+
+    @media (min-width: 768px) {
+        grid-template-columns: repeat(auto-fill, minmax(50px, 1fr));
+        grid-auto-rows: 50px;
+        grid-gap: 12px;
```
Comment (Dammic) [inline src/components/Alphabet/Alphabet.css]: This grid-gap here is unnecessary (it is set for min-width: 576px already)

File: src/components/Alphabet/Alphabet.css [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    display: grid;
+    padding: 12px 20px;
+    margin: 0 auto;
+    max-width: 790px;
+    grid-template-columns: repeat(auto-fill, minmax(45px, 1fr));
+    grid-auto-rows: 40px;
+    grid-gap: 10px;
+
+    @media (min-width: 576px) {
+        grid-template-columns: repeat(auto-fill, minmax(45px, 1fr));
```
Comment (Dammic) [inline src/components/Alphabet/Alphabet.css]: This grid-template-columns is not necessary here (it is set for the mobile and up a few lines above)

File: src/components/Alphabet/Letter/Letter.css [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    font-size: 25px;
+    cursor: pointer; 
+
+    @media (min-width: 576px) {
+        border-radius: 12px;
+        font-size: 30px;
+    }
+
+    @media (min-width: 768px) {
+        border-radius: 12px;
```
Comment (Dammic) [inline src/components/Alphabet/Letter/Letter.css]: This border-radius here is not necessary (it is set for the 576 and up)

File: src/components/GameOver/GameOver.css [hunk 4]
```diff
@@ -0,0 +1,27 @@
+.resultField {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin: 0 auto;
+    width: 500px;
```
Comment (Dammic) [inline src/components/GameOver/GameOver.css]: What about resolutions lower than 500px? For those, the game over will go over the screen. and a horizontal scroll will appear.

File: src/components/GameOver/GameOver.css [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    border: 1px solid #dedede;
+    border-radius: 20px;
+}
+
+.gameOverFrame {
+    font-size: 45px;
+    font-weight: 600;
+    padding: 40px;
+    text-shadow: 1px 1px #000000;
+    color: #2e8b57;
```
Comment (Dammic) [inline src/components/GameOver/GameOver.css]: This color is probably unnecessary? (it is set both by win and loose)

File: src/components/GameOver/GameOver.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { GAME_STATE_FAILED, GAME_STATE_WON } from '../../../consts';
+import styles from './GameOver.css';
+import classnames from 'classnames';
```
Comment (Dammic) [inline src/components/GameOver/GameOver.js]: Please put library imports above components imports (absolute imports before relative imports)

File: src/components/ShowGallows/ShowGallows.css [hunk 7]
```diff
@@ -0,0 +1,30 @@
+.container {
+    position: relative;
+    display: flex;
+    justify-content: center;
+    padding: 20px;
+
+    @media (min-width: 576px) {
+        padding: 20px;
```
Comment (Dammic) [inline src/components/ShowGallows/ShowGallows.css]: Unnecessary media query

File: src/components/ShowGallows/ShowGallows.css [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        font-size: 22px;
+    }
+}
+
+.gallowsImage {
+    height: 100%;
+    width: 100%;
+
+    @media (min-width: 576px) {
+        max-width: 700px;
```
Comment (Dammic) [inline src/components/ShowGallows/ShowGallows.css]: Unnecessary media query ( you can move the max-width: 700px to the .gallowsImage directly, without media query)

File: phrases.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+const phrases = {
+    actor: ['Brad Pitt', 'Tom Cruise', 'Tommy Lee Jones'],
+    ['polish singer']: ['Dawid Podsiadło', 'Szymon Wydra', 'Mieczysław Fogg', 'Czesław Niemen', 'Marek Grechuta'],
```
[the remark this answers is not in the document]
Comment (DominikDR) [reply phrases.js]: Right, I don't know how it get here :smiley:

File: routes/phrasesRoute.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+            const revealedPhrase = revealLetterInPhrase(gameStatus.phrase, gameStatus.encodedPhrase, req.body.letter);
+            gameStatus.encodedPhrase = revealedPhrase;
+            if (gameStatus.encodedPhrase === gameStatus.phrase) gameStatus.endState = GAME_STATE_WON;
+
+            gameStateForClient = pick(gameStatus, gameStateKeys);
+            res.send(gameStateForClient);
+            return;
+        } else {
+            ++gameStatus.failsCounter;
+            if (gameStatus.failsCounter === 6) gameStatus.endState = GAME_STATE_FAILED;
```
[the remark this answers is not in the document]
Comment (DominikDR) [reply routes/phrasesRoute.js]: Moved into variable MAX_ATTEPMTS in const.js :+1:

File: routes/phrasesRoute.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const { gameState, counterID, gameStateKeys, ommitedChars, createNewGame, getRandomPhrase, encryptPhrase, revealLetterInPhrase } = require('../gameServerLogic/gameServerLogic');
+const { GAME_STATE_FAILED, GAME_STATE_WON } = require('../consts');
+const pick = require('lodash.pick');
+
+const routes = express.Router();
+
+routes.use(bodyParser.json());
+routes.use(bodyParser.urlencoded({ extended: true }));
```
[the remark this answers is not in the document]
Comment (DominikDR) [reply routes/phrasesRoute.js]: Yes it was possible. Moved into server.js

**Pull request merged**